rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java

```diff
@@ -777,8 +776,8 @@ public void finishNodeSourceShutdown(Client initiator) {
      * @return a list of alive nodes
      */
     @ImmediateService
-    public LinkedList<Node> getAliveNodes() {
-        LinkedList<Node> nodes = new LinkedList<>();
+    public LinkedList<RMNode> getAliveNodes() {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The return type of this method should be an interface such as "List" rather than the implementation "LinkedList". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1319)